src/codegen/general.rs

```diff
@@ -62,21 +63,26 @@ pub fn define_object_type(
         })
         .collect();
 
+    let class_name = glib_class_name.map_or ("()".to_string(),
+                                             |s| format!("ffi::{}", s));
+
     try!(writeln!(w, ""));
     try!(writeln!(w, "glib_wrapper! {{"));
     if parents.is_empty() {
         try!(writeln!(
             w,
-            "\tpub struct {}(Object<ffi::{}>);",
+            "\tpub struct {}(Object<ffi::{}, {}>);",
             type_name,
-            glib_name
+            glib_name,
+            class_name
         ));
     } else if external_parents {
         try!(writeln!(
             w,
-            "\tpub struct {}(Object<ffi::{}>): [",
+            "\tpub struct {}(Object<ffi::{}, {}>): [",
```

Comment:
You could also implement the macro so that it can have one or two parameters for `Object<..>`. Not sure which is nicer but with having the macro in two variants there would be less noise in the generated code for things that have no class struct

Reply:
You are absolutely correct!  I started with the assumption that everything had a class struct, then it turned out that that wasn't true, I added the (), and left it at that.

But with your scheme, there wouldn't be an API break, just things that cannot be subclassed with gnome-class.

I like the idea of being able to not specify a class struct; I'll change the code to do that.